server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -137,10 +146,21 @@ public Task register(String type, String action, TaskAwareRequest request) {
         } else {
             Task previousTask = tasks.put(task.getId(), task);
             assert previousTask == null;
+            startTrace(threadContext, task);
         }
         return task;
     }
 
+    private void startTrace(ThreadContext threadContext, Task task) {
+        TaskId parentTask = task.getParentTaskId();
+        Map<String, Object> attributes = new HashMap<>();
```

Comment:
We could pre-size the map correctly here or just use `Map.of(` (with a slight bit of repetition for the task id key but whatever) to save some memory and make this compile nicer since its on the hot path.